config/decoder_test.go

```diff
@@ -0,0 +1,300 @@
+			var x int16
+			require.NoError(t, p.Get(Root).Populate(&x))
+			assert.Equal(t, int16(1), x)
+		},
+		"uint16": func(p Provider, t *testing.T) {
+			var x uint16
+			require.NoError(t, p.Get(Root).Populate(&x))
+			assert.Equal(t, uint16(1), x)
+		},
+		"int32": func(p Provider, t *testing.T) {
+			var x int32
+			require.NoError(t, p.Get(Root).Populate(&x))
+			assert.Equal(t, int32(1), x)
+		},
+		"uint32": func(p Provider, t *testing.T) {
+			var x uint32
+			require.NoError(t, p.Get(Root).Populate(&x))
+			assert.Equal(t, uint32(1), x)
+		},
+		"int64": func(p Provider, t *testing.T) {
+			var x int64
+			require.NoError(t, p.Get(Root).Populate(&x))
+			assert.Equal(t, int64(1), x)
+		},
+		"uint64": func(p Provider, t *testing.T) {
+			var x uint64
+			require.NoError(t, p.Get(Root).Populate(&x))
+			assert.Equal(t, uint64(1), x)
+		},
+		"float32": func(p Provider, t *testing.T) {
+			var x float32
+			require.NoError(t, p.Get(Root).Populate(&x))
+			assert.Equal(t, float32(1), x)
+		},
+		"float64": func(p Provider, t *testing.T) {
+			var x float64
+			require.NoError(t, p.Get(Root).Populate(&x))
+			assert.Equal(t, float64(1), x)
+		},
+		"uintptr": func(p Provider, t *testing.T) {
+			var x uintptr
+			require.NoError(t, p.Get(Root).Populate(&x))
+			assert.Equal(t, uintptr(1), x)
+		},
+	}
+
+	for from, provider := range providers {
+		for to, test := range conversions {
+			assert.True(t, t.Run(
+				fmt.Sprintf("From %q to %q", from, to),
+				func(t *testing.T) {
+					test(provider, t)
+				}),
+			)
+		}
+	}
+}
+
+func TestNumericOverflows(t *testing.T) {
+	p := newValueProvider(math.MaxFloat64)
+
+	conversions := map[string]func(p Provider) error{
+		"int": func(p Provider) error {
+			var x int
+			return p.Get(Root).Populate(&x)
+		},
+		"uint": func(p Provider) error {
+			var x uint
+			return p.Get(Root).Populate(&x)
+		},
+		"int8": func(p Provider) error {
+			var x int8
+			return p.Get(Root).Populate(&x)
+		},
+		"uint8": func(p Provider) error {
+			var x uint8
+			return p.Get(Root).Populate(&x)
+		},
+		"int16": func(p Provider) error {
+			var x int16
+			return p.Get(Root).Populate(&x)
+		},
+		"uint16": func(p Provider) error {
+			var x uint16
+			return p.Get(Root).Populate(&x)
+		},
+		"int32": func(p Provider) error {
+			var x int32
+			return p.Get(Root).Populate(&x)
+		},
+		"uint32": func(p Provider) error {
+			var x uint32
+			return p.Get(Root).Populate(&x)
+		},
+		"int64": func(p Provider) error {
+			var x int64
+			return p.Get(Root).Populate(&x)
+		},
+		"uint64": func(p Provider) error {
+			var x uint64
+			return p.Get(Root).Populate(&x)
+		},
+		"float32": func(p Provider) error {
+			var x float32
+			return p.Get(Root).Populate(&x)
+		},
+		"uintptr": func(p Provider) error {
+			var x uintptr
+			return p.Get(Root).Populate(&x)
+		},
+	}
+
+	for to, f := range conversions {
+		assert.True(t, t.Run(fmt.Sprintf("%q overflow", to), func(t *testing.T) {
```

Comment:
They are already subtests :) I was a bit confused about testing.T.Run returning a bool, reporting whether a subtest failed or not, but it looks like it already marks parent:
https://golang.org/src/testing/testing.go?s=15898:15921#L427